packages/embark-webserver/src/index.js

```diff
@@ -16,7 +16,8 @@ class WebServer {
     this.fs = embark.fs;
     this.buildDir = embark.config.buildDir;
     this.webServerConfig = embark.config.webServerConfig;
-    if (!this.webServerConfig.enabled) {
+
+    if (!this.webServerConfig.enabled || !embark.config.pipelineConfig.enabled) {
```

Comment:
This totally does the trick and with the way things are right now, it's probably the most straight forward way to do this. However, you probably notice that we're explicitly expecting a pipelineConfig inside the webserver module, which breaks decoupling.

Ideally, we don't actually have to do these checks in this module at all and don't load/bootstrap it, if it's not needed. Unfortunately, with the design of how configurations are loaded (after the engine has been initialized), this seems the "earliest" when we know whether the pipeline is enabled or not.